aip/general/0236.md

```diff
@@ -0,0 +1,315 @@
+- startPreview is not idempotent and using it twice on the same resource
+  **must** return an error.
+- Etags **must** be used to guarantee resource freshness.
+
+#### stopPreview
+
+```proto
+rpc StopPreviewPolicyExperiment(StopPreviewPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:stopPreview"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "StopPreviewPolicyExperimentMetadata"
+  };
+}
+
+message StopPreviewPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+}
+
+message StopPreviewPolicyExperimentMetadata {}
+```
+- Admin flow API to stop previewing a resource **must** be implemented.
+  - Remains declarative friendly as only an output only field is modified.
+- If stopPreview is called specifically on a resource which is not actively
+  being previewed, then it **must** return an error message.
+- When the API is called, the `state` **must** be set to INACTIVE and
+  `stop_time` **must** be set to the current time.
+- stopPreview is not idempotent, calling stop twice on the same resource
+  **must** return an error.
+- Etags **must** be used to guarantee resource freshness.
+
+#### commit
+
+A new API called commit **may** be implemented to promote the experimental
+configuration of a resource to the effective version and perform necessary
+cleanup. It will copy the fields-under-test from the experiment into the live
+policy and then delete the experiment.
+
+This is a convenience API which is not declarative friendly. The alternative
+supported for declarative clients is to manually copy fields from an experiment
+onto the live Policy then delete the experiment.
+
+```proto
+rpc CommitPolicyExperiment(CommitPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:commit"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "CommitPolicyExperimentMetadata"
+  };
+}
+
+message CommitPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+  string parent_etag = 3; // optional.
+}
+
+message CommitPolicyExperimentMetadata {}
+```
+- Commit **must** atomically copy the fields-under-test from the experiment into
+  the live Policy then deletes the experiment.
+- If any experiment fails to be committed, the preview **must** be continued and
+  the Policy **must** not be updated.
+- Commit is not idempotent and calling commit twice on the same experiment
+  **must** return an error.
+- Commit **may** be called at any point of time for an experiment resource being
+  used for preview, either when the logs are actively being generated or after
+  they are stopped. It has the same behavior in each case.
+- Etags **must** be used to guarantee resource freshness.
```

Comment:
Describe what the connotation of an "etag" is in the context of the case of invoking this method. Don't capitalize "etag" and throughout.